lib/fshash/metadata.go

```diff
@@ -0,0 +1,118 @@
+package fshash
+
+import (
+	"archive/tar"
+	"io"
+	"os"
+	"path/filepath"
+	"sort"
+	"time"
+
+	"github.com/spacemonkeygo/errors"
+	"github.com/ugorji/go/codec"
+	"polydawn.net/repeatr/def"
+)
+
+type Metadata tar.Header
+
+func ReadMetadata(path string, optional ...os.FileInfo) Metadata {
+	var fi os.FileInfo
+	var err error
+	if len(optional) > 0 {
```

Comment:
Ah, I think I had some intention to refactor other uses outside of tests to take this path too and forgot about it.  Doing that now.  (varargs are because there's no way to specify a single optional arg, womp womp polymorphism.)